### PR TITLE
Update go-client

### PR DESCRIPTION
**Pull request by michaljurecko**
See: 
- https://github.com/keboola/go-client/pull/48
- https://github.com/keboola/go-client/pull/49

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -23,7 +23,7 @@ require (
 	github.com/jarcoal/httpmock v1.2.0
 	github.com/joho/godotenv v1.4.0
 	github.com/jpillora/longestcommon v0.0.0-20161227235612-adb9d91ee629
-	github.com/keboola/go-client v0.14.0
+	github.com/keboola/go-client v0.15.0
```
Comment (michaljurecko) [inline go.mod:26]: Update.

File: internal/pkg/state/remote/manager.go [hunk 2]
```diff
@@ -335,6 +335,8 @@ func (u *UnitOfWork) createRequest(objectState model.ObjectState, object model.O
 			if errors.As(err, &storageApiErr) {
 				if storageApiErr.ErrCode == "configurationAlreadyExists" || storageApiErr.ErrCode == "configurationRowAlreadyExists" {
 					// Object exists -> update instead of create
+					// This can happen if there is a disconnected "variables" configuration, and push connects it again.
+					// See TestCliE2E/push/variables-add-relation
 					return u.updateRequest(objectState, object, recipe, nil).SendOrErr(ctx, sender)
```
Comment (michaljurecko) [inline internal/pkg/state/remote/manager.go:340]: Tu som doplnil komentar.
Handlovanie `configurationAlreadyExists` mame aj tu v KAC, ale nie pre chybu co sme riesili ako workaround v `go-client`. 

Jedna sa o inu situaciu, ked `variables` konfiguraciu, ktora nie je napojena na ziadny config ignorujeme, pri nacitani remote state.

Potom tam mame test na edge case `TestCliE2E/push/variables-add-relation`, ze by niekto pripojit tuto EXISTUJUCU variables konfiguraciu k nejake konfiguracii (ktora tie variables pouziva). Kedze na zaciatku tie `variables` ignorujeme (lebo su odpojene), tak neskor uz (v aktualnom kode) nevieme zistit, ze v remote projekte uz existuju.

TLDR: pridany koment, na co je tento kod

**Review by JakubMatejka — APPROVED** [review]
